Source/Core/Core/FifoPlayer/FifoPlayer.cpp

```diff
@@ -103,6 +103,9 @@ bool FifoPlayer::Play()
 				++m_CurrentFrame;
 			}
 		}
+
+		// Let the waiting thread know we are done leaving
+		PowerPC::FinishStateMove();
```

Comment:
This does still have the unfortunate effect that if we aren't looping the fifo log, then PowerPC::Stop will always wait 1 second before giving up and leaving.